gtdynamics/kinematics/Kinematics.h

```diff
@@ -131,29 +178,60 @@ class Kinematics : public Optimizer {
   EqualityConstraints pointGoalConstraints(
       const CONTEXT& context, const ContactGoals& contact_goals) const;
 
+  /**
+   * @fn Create a pose prior for a given link for each given pose.
+   * @param context Slice or Interval instance.
+   * @param pose_goals goals for poses
```

Comment:
You are still not explaining what pose_goals is. size_t can be many things, what does it mean? Does it pertain to all contexts?